server/app/analytics/job.py

```diff
@@ -0,0 +1,93 @@
+from google.appengine.ext import ndb, deferred
+from app.analytics.mapper import Mapper
+from app.models import Base
+
+
+class JobMapper(Mapper):
+    """
+    A wrapper class for the NDB Mapper that stores
+    map results in an NDB AnalyticsDump in addition to just mapping.
+    """
+    def __init__(self, kind, job_dump, job_mapper, filters):
+        super(JobMapper, self).__init__()
+        self.kind = kind
+        self.job_dump = job_dump
+        self.job_mapper = job_mapper
+        self.set_filters(filters)
+
+    def map(self, entity):
+        try:
+            map_result = self.job_mapper(entity)
+        except:
+            raise deferred.PermanentTaskFailure()
+        self.job_dump.map_result.append(map_result)
```

Comment:
Oh, whoops, didn't read the description.

Maybe we should make this deletion easier? Or make it optional to not output map results? If there's a common MapReduce job we run, I can imagine not wanting to have to save the output a lot. And that would save on datastore stuff.